File: python_solutions/problem_4.py
def main():

    palindromes = []

    for i in range(1, 1000):
        for j in range(1, 1000):
            prod = i * j

            if str(prod) == str(prod)[::-1]:
                palindromes.append(prod)

    return max(palindromes)


if __name__ == "__main__":
    print(main())

File: solutions/csharp/Problem4.cs
using System.Numerics;

namespace ProjectEuler
{
    class Problem4 : Problem
    {
        public override BigInteger Solve()
        {
            int ceiling = 1000;
            List<int> palindromes = new();

            for (int i = 1; i < ceiling; i++)
            {
                for (int j = 1; j < ceiling; j++)
                {
                    int prod = i * j;

                    if (EulerLib.IsPalindrome(prod))
                    {
                        palindromes.Add(prod);
                    }
                }
            }

            return palindromes.Max();
        }
    }
}